test/libsolidity/semanticTests/literal_suffixes/intermediate.sol

```diff
@@ -0,0 +1,21 @@
+type Length is uint;
+
+function km(uint x) pure returns (Length) { return Length.wrap(x * 1000); }
+
+struct Float {
+    uint mantissa;
+    int exponent;
+}
+
+function f(uint mat, int exp) pure returns (Float memory) {
```

Comment:
I wonder if this should be mantissa and exponent, or numerator and denominator.

But yeah, this example looks nice!

Reply:
The problem is that for converting numerator and denominator to mantissa and exponent, in general you need a higher precision... if we had arbitrary-precision compile-time-constant expression evaluation we could easily make it numerator/denominator or simply ``rational`` - but mantissa + exponent is the significantly quicker route to get fixed point literals...

Reply:
My main point is that the input is really in mantissa + exponent notation and not in numerator+denominator notation. You can apply this function only to literals (`1.34`, `-3.4e20`)  and not to rational expressions.

Reply:
Rational literals *are* floating point in that sense - they do have a point at arbitrary position :-).
And ``(2**256 - 1) / 10**76`` is not a user-defined literal...

Reply:
As discussed on the call, the exponent is always negative (because otherwise, the integer suffix function would be called). 
Because of that, there is no need for branching.
Since signed types for `**` are unsupported, it might be better to use "mantissa + digits", i.e. negate the exponent:
```solidity
function f18(uint mantissa, int8 digits) returns (Fixed18) {
  return Fixed18.wrap(mantissa * 10 ** (18 - digits);
}
```
Even if we had a branch, this should all be fully inlined/specialized because of the constants everywhere.